src/routes/joinGame.ts

```diff
@@ -61,56 +63,62 @@ router.get(
-                startingFunds: 0,
-                players: [],
-                gameMasterHash: sha256(result.rows[0].game_master).toString(),
-              }
-              await client
-                .query(getGameInfoQuery, getGameInfoValues)
-                .then((result) => {
-                  gameInfo.smallBlind = parseInt(result.rows[0].small_blind)
-                  gameInfo.startingFunds = parseInt(
-                    result.rows[0].starting_funds
-                  )
-                })
-              await client
-                .query(getPlayersInRoomQuery, getPlayersInRoomValues)
-                .then((result) => {
-                  const message = {
-                    data: {
-                      type: 'playerJoined',
-                      // We know that the nickname will be defined
-                      // because we checked it with celebrate.
-                      nickname: req.query.nickname as string,
-                      playerHash: sha256(req.query.playerToken).toString(),
-                    },
-                    token: '',
-                  }
-                  result.rows.forEach(async (row) => {
-                    gameInfo.players.push({
-                      nickname: row.nickname,
-                      playerHash: sha256(row.token).toString(),
-                    })
-                    // Sending firebase message to all players except the one
-                    // who just joined.
-                    if (row.token !== req.query.playerToken) {
-                      message.token = row.token
-                      await sendFirebaseMessage(message)
-                    }
-                  })
-                })
-              await client.query(createPlayerQuery, createPlayerValues)
-              res.send(gameInfo)
-            }
+          .query(getGameInfoQuery, getGameInfoValues)
+          .then((result) => {
+            gameInfo.smallBlind = parseInt(result.rows[0].small_blind)
+            gameInfo.startingFunds = parseInt(result.rows[0].starting_funds)
+          })
+
+        // Notify players about new player
+        const playersInRoomResult = await client.query(
+          getPlayersInRoomQuery,
+          getPlayersInRoomValues
+        )
+
+        const message = {
+          data: {
+            type: 'playerJoined',
+            // We know that the nickname will be defined
+            // because we checked it with celebrate.
+            nickname: req.query.nickname as string,
+            playerHash: sha256(req.query.playerToken).toString(),
+          },
+          token: '',
+        }
+        playersInRoomResult.rows.forEach(async (row) => {
+          gameInfo.players.push({
+            nickname: row.nickname,
+            playerHash: sha256(row.token).toString(),
           })
+          // Sending firebase message to all players except the one
+          // who just joined.
+          if (row.token !== req.query.playerToken) {
```

Comment:
Is this if still necessary? We recently changed that the new player is added to the database after sending firebase messages.

Reply:
Fixed it, thanks a lot for the thorough review!